### PR TITLE
Feature/models

### PR DESCRIPTION
**Pull request by arvindgkmit**

### CODE REVIEW

File: models/taskcomment.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+const {
+  Model
+} = require('sequelize');
+module.exports = (sequelize, Sequelize) => {
+  class TaskComment extends Model {
```
Comment (dhruvildave007) [inline models/taskcomment.js]: naming of the file should be either TaskComment, User or task_comment or taskComment.

File: models/user.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      this.belongsToMany(models.Workspace, {
+        through: models.UserWorkspaceMapping,
+        foreignKey: 'userId',
+      });
+
+      this.belongsToMany(models.Sprint, {
+        foreignKey: 'userId',
+        targetKey: 'id'
+      });
+
```
Comment (dhruvildave007) [inline models/user.js]: User should be belongToMany with Workspace, Sprint should be belong to a WorkSpace

File: models/userdesignationmapping.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      // define association here
+      this.belongsTo(models.User, {
+        foreignKey: 'userId'
+      });
+
+      this.belongsTo(models.Designation, {
+        foreignKey: 'designationId'
```
Comment (dhruvildave007) [inline models/userdesignationmapping.js]: add source key and alias also
for eg:
```
this.hasMany(models.Designation, {
        foreignKey: 'user_id',
        sourceKey: 'id',
        as: 'Designation'
      });```

File: models/userrolemapping.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    static associate(models) {
+      // define association here
+       this.belongsTo(models.User, {
+        foreignKey: 'userId'
+      });
+      this.belongsTo(models.Role, {
+        foreignKey: 'roleId'
+      });
+    }
```
Comment (dhruvildave007) [inline models/userrolemapping.js]: same as above comment add sourceKey and as in all models

File: models/workspace.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      },
+      description: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        isAlpha: true
+      },
+      deadline: {
+        type: Sequelize.DATE,
+        allowNull: false,
+      },
```
Comment (dhruvildave007) [inline models/workspace.js]: workspace should not have deadline column

File: models/sprint.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        targetKey: 'id'
+      });
+       this.belongsTo(models.Tag, {
+        foreignKey: 'tagId',
+        targetKey: 'id'
+      });
+    }
+  }
+  Sprint.init({
+    name: {
```
Comment (dhruvildave007) [inline models/sprint.js]: check alignment of your code.
use `prettier` to format your code before commiting

File: models/sprint.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      // define association here
+       this.hasMany(models.Task, {
+        foreignKey: 'sprintId',
+        targetKey: 'id'
+      });
+       this.belongsTo(models.Tag, {
+        foreignKey: 'tagId',
```
Comment (dhruvildave007) [inline models/sprint.js]: `Sprint` should not belong to tags.

File: models/tag.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    static associate(models) {
+      // define association here
+      this.hasMany(models.Sprint, {
+        foreignKey: 'tagId',
+        targetKey: 'id'
+      });
+      this.belongsTo(models.Workspace, {
+        foreignKey: 'workspaceId',
+        targetKey: 'id'
+      });
```
Comment (dhruvildave007) [inline models/tag.js]: tags should belong to Task not workspace

File: models/tag.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        allowNull: false,
+        isAlpha: true
+      },
+      description: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        isAlpha: true
+      },
+      deadline: {
+        type: Sequelize.DATE,
```
Comment (dhruvildave007) [inline models/tag.js]: no need to add deadline in tags.

File: models/task.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        targetKey: 'id'
+      });
+
+      this.belongsTo(models.User, {
+        foreignKey: 'userId',
+        targetKey: 'id'
+      });
+    }
+  }
+  Task.init({
```
Comment (dhruvildave007) [inline models/task.js]: add more columns like description, owners(like who is watching the task), deadline date etc.

File: models/usersprintmapping.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
+'use strict';
```
Comment (dhruvildave007) [inline models/usersprintmapping.js]: no need for this mapping as we are doing User --> WorkSpace mapping

**Review by dhruvildave007 — CHANGES_REQUESTED** [review]
please do required changes

File: models/taskcomment.js (hunk 1 above)
Comment (arvindgkmit) [reply models/taskcomment.js]: ok

File: models/user.js (hunk 2 above)
Comment (arvindgkmit) [reply models/user.js]: ok

File: models/userdesignationmapping.js (hunk 3 above)
Comment (arvindgkmit) [reply models/userdesignationmapping.js]: ok

File: models/workspace.js (hunk 5 above)
Comment (arvindgkmit) [reply models/workspace.js]: ok

File: models/sprint.js (hunk 7 above)
Comment (arvindgkmit) [reply models/sprint.js]: ok

File: models/tag.js (hunk 8 above)
Comment (arvindgkmit) [reply models/tag.js]: ok

File: models/tag.js (hunk 9 above)
Comment (arvindgkmit) [reply models/tag.js]: ok

File: models/task.js (hunk 10 above)
Comment (arvindgkmit) [reply models/task.js]: ok

File: models/usersprintmapping.js (hunk 11 above)
Comment (arvindgkmit) [reply models/usersprintmapping.js]: ok

**Review by dhruvildave007 — APPROVED** [review]

**Pull request merged**